src/ripple/basics/Slice.h

```diff
@@ -131,9 +131,13 @@ inline
 bool
 operator== (Slice const& lhs, Slice const& rhs) noexcept
 {
-    return lhs.size() == rhs.size() &&
-        std::memcmp(
-            lhs.data(), rhs.data(), lhs.size()) == 0;
+    if (lhs.size() != rhs.size())
+        return false;
+
+    if (lhs.size() == 0)
+        return true;
+
+    return std::memcmp(lhs.data(), rhs.data(), lhs.size()) == 0;
```

Comment:
I find that compound `return` to be much harder to read and "visually" inspect for correctness. Also, a quick look at godbolt shows that the code generated is identical, so I'm inclined to leave it as is, unless someone feels very strongly that the alternative is better.